index.html

```diff
@@ -705,31 +710,68 @@
                                 if (cellvar.model.in.height == null) {
                                     inputPort = 0;
-                                } else {
+                                }
+                                else {
                                     inputPort = cellvar.model.in.height;
                                 }
                                 if (cellvar.model.out.height == null) {
                                     outputPort = 0;
-                                } else {
+                                }
+                                else {
                                     outputPort = cellvar.model.out.height;
                                 }
                                 if (cellvar.model.evtin.height == null) {
                                     controlPort = 0;
-                                } else {
+                                }
+                                else {
                                     controlPort = cellvar.model.evtin.height;
                                 }
                                 if (cellvar.model.evtout.height == null) {
                                     commandPort = 0;
-                                } else {
+                                }
+                                else {
                                     commandPort = cellvar.model.evtout.height;
                                 }
+
+                                var flip = false, mirror = false;
+                                var style = cell.style;
+                                var styleObject = styleToObject(style);
+                                if(styleObject['stencilFlipV'] == null){
+                                    flip = false;
+                                }
+                                else
+                                {
+                                    
+                                    if(styleObject['stencilFlipV'] == 0){
+                                        flip = false;
+                                    }
+                                    else{
+                                        flip = true;
+                                    }
+
+                                }
+                                if(styleObject['stencilFlipH'] == null){
+                                    mirror = false;
+                                }
+                                else
+                                {
+                                   
+                                    if(styleObject['stencilFlipH'] == 0 ){
+                                        mirror = false;
+                                    }
+                                    else{
+                                        mirror = true;
+                                    }
+
+                                }
+                                style = objectToStyle(styleObject);
```

Comment:
Whole evening spent in wondering why the script stopped when I used that
function. Anyway now everything works fine. Adhitya I have removed that if
condition and just put one condition.

On 13 July 2016 at 00:25, Adhitya Kamakshidasan <notifications@github.com>
wrote:

> In index.html
> <https://github.com/FOSSEE/xcos-on-web/pull/170#discussion_r70499860>:
>
> > +                                }
> > +                                if(styleObject['stencilFlipH'] == null){
> > +                                    mirror = false;
> > +                                }
> > +                                else
> > +                                {
> > +
> > +                                    if(styleObject['stencilFlipH'] == 0 ){
> > +                                        mirror = false;
> > +                                    }
> > +                                    else{
> > +                                        mirror = true;
> > +                                    }
> > +
> > +                                }
> > +                                style = objectToStyle(styleObject);
>
> @Pooja-Mahadev-Soundalgekar
> <https://github.com/Pooja-Mahadev-Soundalgekar> just punched @jiteshjha
> <https://github.com/jiteshjha> in the face, with her previous comment.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/FOSSEE/xcos-on-web/pull/170/files/ce05ba7adb6aa74508924360660ae442d3e2e5bd#r70499860>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ALzP2qpk9_s5EC0zR4Rnn6vcT1ptX5Ikks5qU-MrgaJpZM4JKJM6>
> .
>